refinery/core/views.py

```diff
@@ -207,8 +207,16 @@ def data_set_slug(request, slug):
 def data_set(request, data_set_uuid, analysis_uuid=None):
     data_set = get_object_or_404(DataSet, uuid=data_set_uuid)
     public_group = ExtendedGroup.objects.public_group()
-
-    if not request.user.has_perm('core.read_meta_dataset', data_set):
+    try:
+        perms = request.user.has_perm('core.read_meta_dataset', data_set)
+    except User.DoesNotExist:
```

Comment:
If the user is not logged in, they are the anon user. What's throwing the exception exactly?